pygments/lexers/dsls.py

```diff
@@ -188,84 +188,174 @@ class ThriftLexer(RegexLexer):
+
+            # IPv6 Address (not 100% correct: that takes more effort)
+            (r'\[([0-9a-fA-F]{0,4}:){2,7}([0-9a-fA-F]{0,4})?((25[0-5]|2[0-4][0-9]|[0-1][0-9]{2}|[0-9]{1,2})\.(25[0-5]|2[0-4][0-9]|[0-1][0-9]{2}|[0-9]{1,2})\.(25[0-5]|2[0-4][0-9]|[0-1][0-9]{2}|[0-9]{1,2})\.(25[0-5]|2[0-4][0-9]|[0-1][0-9]{2}|[0-9]{1,2}))?\]', Number),
+
+            # Numeric
+            (r'\b0[xX]' + _hex + r'+\b', Number.Hex),
+            (r'\b' + _float + r'\s*(day|hr|min|sec|msec|usec)s?\b', Number.Float),
+            (r'\b' + _float + r'\b', Number.Float),
+            (r'\b(\d+)\b', Number.Integer),
+
             # Hostnames
             (_h + r'(\.' + _h + r')+', String),
-            # Numeric
-            (_float + r'\s+(day|hr|min|sec|msec|usec)s?\b', Literal.Date),
-            (r'0[xX]' + _hex, Number.Hex),
-            (_float, Number.Float),
-            (r'\d+', Number.Integer),
-            (r'/', String.Regex, 'regex'),
-            (r'"', String, 'string'),
-            # Operators
-            (r'[!%*/+:<=>?~|-]', Operator),
+        ],
+
+        'operators': [
+            (r'[!%*/+<=>~|&^-]', Operator),
             (r'([-+=&|]{2}|[+=!><-]=)', Operator),
-            (r'(in|match)\b', Operator.Word),
-            (r'[{}()\[\]$.,;]', Punctuation),
-            # Identfier
-            (r'([_a-zA-Z]\w*)(::)', bygroups(Name, Name.Namespace)),
+            (r'\b(in|as|is|of)\b', Operator.Word),
+            (r'\??\$', Operator),
+            # Technically, colons are often used for punctuation/separation.
+            # E.g. field name/type separation.
+            (r'[?:]', Operator),
+        ],
+
+        'punctuation': [
+            (r'\?\$', Punctuation),
+            (r'[{}()\[\],;:.]', Punctuation),
+        ],
+
+        'identifiers': [
+            (r'([a-zA-Z_]\w*)(::)', bygroups(Name, Punctuation)),
             (r'[a-zA-Z_]\w*', Name)
         ],
+
         'string': [
+            (r'\\.', String.Escape),
+            (r'%-?[0-9]*(\.[0-9]+)?[DTdxsefg]', String.Escape),
             (r'"', String, '#pop'),
-            (r'\\([\\abfnrtv"\']|x[a-fA-F0-9]{2,4}|[0-7]{1,3})', String.Escape),
-            (r'[^\\"\n]+', String),
-            (r'\\\n', String),
-            (r'\\', String)
+            (r'.', String),
         ],
+
         'regex': [
+            (r'\\.', String.Escape),
             (r'/', String.Regex, '#pop'),
-            (r'\\[\\nt/]', String.Regex),  # String.Escape is too intense here.
-            (r'[^\\/\n]+', String.Regex),
-            (r'\\\n', String.Regex),
-            (r'\\', String.Regex)
-        ]
+            (r'.', String.Regex),
+        ],
     }
 
 
+class BroLexer(ZeekLexer):
+    """
+    For `Zeek <https://www.zeek.org/>`_ scripts.
+    Note, this is equivalent to ZeekLexer (Bro is the old name for Zeek).
```

Comment:
I think this should be done by moving the alias/filename pattern to ZeekLexer, and just having a `BroLexer = ZeekLexer` for compatibility.

Reply:
I've hit a point where I may need to modify a unit test to get that way to work, so could you help take a look at what you'd like done here or else point me at another example where this has been done?

Here's the patch I tried:

```diff
diff --git a/pygments/lexers/_mapping.py b/pygments/lexers/_mapping.py
index 92aaca39..5405785f 100644
--- a/pygments/lexers/_mapping.py
+++ b/pygments/lexers/_mapping.py
@@ -65,7 +65,7 @@ LEXERS = {
     'BooLexer': ('pygments.lexers.dotnet', 'Boo', ('boo',), ('*.boo',), ('text/x-boo',)),
     'BoogieLexer': ('pygments.lexers.verification', 'Boogie', ('boogie',), ('*.bpl',), ()),
     'BrainfuckLexer': ('pygments.lexers.esoteric', 'Brainfuck', ('brainfuck', 'bf'), ('*.bf', '*.b'), ('application/x-brainfuck',)),
-    'BroLexer': ('pygments.lexers.dsls', 'Bro', ('bro',), ('*.bro',), ()),
+    'BroLexer': ('pygments.lexers.dsls', 'Zeek', ('zeek', 'bro'), ('*.zeek', '*.bro'), ()),
     'BugsLexer': ('pygments.lexers.modeling', 'BUGS', ('bugs', 'winbugs', 'openbugs'), ('*.bug',), ()),
     'CAmkESLexer': ('pygments.lexers.esoteric', 'CAmkES', ('camkes', 'idl4'), ('*.camkes', '*.idl4'), ()),
     'CLexer': ('pygments.lexers.c_cpp', 'C', ('c',), ('*.c', '*.h', '*.idc'), ('text/x-chdr', 'text/x-csrc')),
@@ -474,7 +474,7 @@ LEXERS = {
     'XtlangLexer': ('pygments.lexers.lisp', 'xtlang', ('extempore',), ('*.xtm',), ()),
     'YamlJinjaLexer': ('pygments.lexers.templates', 'YAML+Jinja', ('yaml+jinja', 'salt', 'sls'), ('*.sls',), ('text/x-yaml+jinja', 'text/x-sls')),
     'YamlLexer': ('pygments.lexers.data', 'YAML', ('yaml',), ('*.yaml', '*.yml'), ('text/x-yaml',)),
-    'ZeekLexer': ('pygments.lexers.dsls', 'Zeek', ('zeek',), ('*.zeek',), ()),
+    'ZeekLexer': ('pygments.lexers.dsls', 'Zeek', ('zeek', 'bro'), ('*.zeek', '*.bro'), ()),
     'ZephirLexer': ('pygments.lexers.php', 'Zephir', ('zephir',), ('*.zep',), ()),
     'ZigLexer': ('pygments.lexers.zig', 'Zig', ('zig',), ('*.zig',), ('text/zig',)),
 }
diff --git a/pygments/lexers/dsls.py b/pygments/lexers/dsls.py
index d88bac99..5894606f 100644
--- a/pygments/lexers/dsls.py
+++ b/pygments/lexers/dsls.py
@@ -195,8 +195,8 @@ class ZeekLexer(RegexLexer):
     .. versionadded:: 2.5
     """
     name = 'Zeek'
-    aliases = ['zeek']
-    filenames = ['*.zeek']
+    aliases = ['zeek', 'bro']
+    filenames = ['*.zeek', '*.bro']
 
     _hex = r'[0-9a-fA-F]'
     _float = r'((\d*\.?\d+)|(\d+\.?\d*))([eE][-+]?\d+)?'
@@ -345,16 +345,7 @@ class ZeekLexer(RegexLexer):
     }
 
 
-class BroLexer(ZeekLexer):
-    """
-    For `Zeek <https://www.zeek.org/>`_ scripts.
-    Note, this is equivalent to ZeekLexer (Bro is the old name for Zeek).
-
-    .. versionadded:: 1.5
-    """
-    name = 'Bro'
-    aliases = ['bro']
-    filenames = ['*.bro']
+BroLexer = ZeekLexer
 
 
 class PuppetLexer(RegexLexer):
```

Here's the failing unit test:

```
    def test_get_lexers():
        # test that the lexers functions work
        for func, args in [(lexers.get_lexer_by_name, ("python",)),
                           (lexers.get_lexer_for_filename, ("test.py",)),
                           (lexers.get_lexer_for_mimetype, ("text/x-python",)),
                           (lexers.guess_lexer, ("#!/usr/bin/python -O\nprint",)),
                           (lexers.guess_lexer_for_filename, ("a.py", "<%= @foo %>"))
                           ]:
            x = func(opt='val', *args)
            assert isinstance(x, lexers.PythonLexer)
            assert x.options["opt"] == "val"
    
        for cls, (_, lname, aliases, _, mimetypes) in lexers.LEXERS.items():
>           assert cls == lexers.find_lexer_class(lname).__name__
E           AssertionError: assert 'BroLexer' == 'ZeekLexer'
E             - BroLexer
E             + ZeekLexer

tests/test_basic_api.py:124: AssertionError
```

Except that test does sound legit the way it is: if `BroLexer` remains in `LEXERS` at all (think it needs to for compat), then it's going to need a unique `name` attribute ?

Reply:
I don't think the old one needs to stay in `LEXERS`, as long as the old aliases and filenames work with the new one...

Reply:
But the `_mapping.py` file containing the `LEXERS` definition auto-generates itself, so either need to change the generation code to ignore `BroLexer` somehow or else move it to a different place where that logic won't find it?

Reply:
Just needs to be removed from `__all__`.

Reply:
Thanks, sounds good, new commit does that.  Let me know if there's anything further.